lib/msal-angular/src/msal.guard.ts

```diff
@@ -137,27 +138,31 @@ export class MsalGuard {
      * If a page with MSAL Guard is set as the redirect for acquireTokenSilent,
      * short-circuit to prevent redirecting or popups.
      */
-    if (typeof window !== "undefined") {
-      if (
-        UrlString.hashContainsKnownProperties(window.location.hash) &&
-        BrowserUtils.isInIframe() &&
-        !this.authService.instance.getConfiguration().system
-          .allowRedirectInIframe
-      ) {
-        this.authService
-          .getLogger()
-          .warning(
-            "Guard - redirectUri set to page with MSAL Guard. It is recommended to not set redirectUri to a page that requires authentication."
-          );
-        return of(false);
-      }
-    } else {
+    if (typeof window === "undefined") {
       this.authService
         .getLogger()
         .info(
           "Guard - window is undefined, MSAL does not support server-side token acquisition"
         );
       return of(true);
+    } else {
+      try {
+        BrowserUtils.blockReloadInHiddenIframes();
+      } catch (error) {
+        if (
+          error instanceof BrowserAuthError &&
+          error.errorCode === BrowserAuthErrorCodes.blockIframeReload &&
```

Comment:
This check is redundant as `blockReloadInHiddenIframes` will only ever throw this error.